src/optimizer/README.md

```diff
@@ -1,34 +1,68 @@
 # regexp-tree: Optimizer module
 
-Optimizer transforms your regexp into an _optimized_ version, replacing some sub-expressions with their idiomatic patterns. This might be good for different kinds of minifiers, as well as for regexp machines.
+## Overview
+
+Optimizer transforms your regexp into an _optimized_ version, replacing some sub-expressions with their idiomatic patterns.
+
+Advantages:
+- Optimized regexps are smaller -- good for minification.
+- Optimized regexps may be easier to read.
+- Some optimizations will reduce your risk of REDOS.
 
 Example:
 
 ```js
 /[a-zA-Z_0-9][A-Z_\da-z]*\e{1,}/
 ```
 
-Is transformed into:
+becomes:
 
 ```js
 /\w+e+/
 ```
 
-**Optimizations applied:**
-
-- `aa*` -> `a+`
-- `()`, `(?:)` - `<empty>` (remove empty groups)
-- `a{1,}` -> `a+`
-- `a{1}` -> `a`
-- `a{3,3}` -> `a{3}`
-- `[a-zA-Z_0-9]` -> `\w`
-- `[^a-zA-Z_0-9]` -> `\W`
-- `[\d\d]` -> `[\d]` (remove duplicates from char-class)
-- `[\d]` -> `\d` (remove char class for single char)
-- `[0-9]` -> `\d`
-- `[^0-9]` -> `\D`
-- `[ \t\r\n\f]` -> `\s`
-- `[^ \t\r\n\f]` -> `\S`
-- `\e` -> `e` (remove unnecessary escape)
-- `(?:a|b|c)` -> `[abc]`
-- ...
+## API
+
+`optimize(regexp, [transformsWhitelist])`: Optimize the regexp. Optionally request specific transforms.
+
+Transforms will be applied until no further optimization progress is made.
+
+If you wish to specify a whitelist, give an array of transform names from the table below.
+
+```js
+const optimizer = require('./index.js');
+const inefficient = /[0-9]/;
+
+const optimized1 = optimizer.optimize(inefficient);
+const optimized2 = optimizer.optimize(inefficient, [
+	'charClassToMeta',       // [0-9] -> [\d]
+	'charClassToSingleChar', // [\d] -> \d
```

Comment:
please use 2 spaces in examples, and development

(for development you can also setup Prettier plugin, which automatically formats code)